src/tmx/TmxUtils/src/SNMPClient.cpp

```diff
@@ -97,6 +96,156 @@ namespace tmx::utils
         }
     }
 
+    // Client defaults to SNMPv3
+    snmp_client::snmp_client(const std::string &ip, const int &port, const std::string &community,
+                             const std::string &snmp_user, const std::string &securityLevel, const std::string &authProtocol, const std::string &authPassPhrase, const std::string &privProtocol, const std::string &privPassPhrase, int snmp_version, int timeout)
+
+        : ip_(ip), port_(port), community_(community), snmp_version_(snmp_version), timeout_(timeout)
+    {
+
+        PLOG(logDEBUG1) << "Starting SNMP Client. Target device IP address: " << ip_ << ", Target device SNMP port: " << port_;
+
+        // Bring the IP address and port of the target SNMP device in the required form, which is "IPADDRESS:PORT":
+        std::string ip_port_string = ip_ + ":" + std::to_string(port_);
+        char *ip_port = &ip_port_string[0];
+
+        // Initialize SNMP session parameters
+        init_snmp("snmp_init");
+        snmp_sess_init(&session);
+        session.peername = ip_port;
+        session.version = snmp_version_; // SNMP_VERSION_3
+        session.securityName = (char *)snmp_user.c_str();
+        session.securityNameLen = snmp_user.length();
+        session.securityModel = SNMP_SEC_MODEL_USM;
+
+        // Fallback behavior to setup a community for SNMP V1/V2
+        if (snmp_version_ != SNMP_VERSION_3)
+        {
+            session.community = (unsigned char *)community.c_str();
+            session.community_len = community_.length();
+        }
+
+        // Set security level
+        if (securityLevel == "authPriv")
+        {
+            session.securityLevel = SNMP_SEC_LEVEL_AUTHPRIV;
+        }
+
+        else if (securityLevel == "authNoPriv")
+        {
+            session.securityLevel = SNMP_SEC_LEVEL_AUTHNOPRIV;
+        }
+
+        else
+            session.securityLevel = SNMP_SEC_LEVEL_NOAUTH;
+
+        // Defining and generating authentication config
+        oid *usmAuthProto;
```

Comment:
Is it possible to put this in a separate function?

Reply:
What would be our justification for removing this from the constructor?

Reply:
It's about readability of this constructor. I am suggesting move this logic ,usmAuthProto assignment, to a separate function and call from the constructor.